### PR TITLE
[Flat storage]  adding debug tool for flat storage

### PR DESCRIPTION
**Pull request by mm-near**
Couple basic debug commands for flat storage:
- viewing current state
- removing flat storage
- creating flat storage
- verification (comparing with trie)

Example usage (verify shard 0):
```
target/debug/neard flat-storage verify 0
```

Currently it looks only on the main column (it doesn't display any information about the deltas)

### CODE REVIEW

File: tools/flat-storage/src/flat.rs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                        println!(
+                            "Different keys {:?} in trie, {:?} in flat. ",
+                            item_trie.0, item_flat.0
+                        );
+                        success = false;
+                        break;
+                    }
+                    if item_trie.1.len() != value_ref.length as usize {
+                        println!(
+                            "Wrong values for key: {:?} length trie: {:?} vs {:?}",
```
Comment (Longarithm) [inline tools/flat-storage/src/flat.rs:242]:
```suggestion
                            "Wrong `ValueRef::length` for key: {:?} in trie: {:?} vs. flat storage: {:?}",
```

I couldn't fully understand how to parse it, so I'm leaving my suggestion here and below

File: tools/flat-storage/src/flat.rs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                            item_trie.1.len(),
+                            value_ref.length
+                        );
+                        success = false;
+                        break;
+                    }
+
+                    if near_primitives::hash::hash(&item_trie.1) != value_ref.hash {
+                        println!(
+                            "Wrong hash for key: {:?} length trie: {:?} vs {:?}",
```
Comment (Longarithm) [inline tools/flat-storage/src/flat.rs:253]:
```suggestion
                            "Wrong `ValueRef::hash` for key: {:?} in trie: {:?} vs. flat storage: {:?}",
```

File: tools/flat-storage/src/flat.rs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    /// View the current state of flat storage
+    View,
+
+    /// Reset the flat storage state (remove all the contents)
+    Reset(ResetCmd),
+
+    /// Init the flat storage state, by copying from trie
+    Init(InitCmd),
+
+    /// Verify flat storage state
```
Comment (Longarithm) [inline tools/flat-storage/src/flat.rs:35]: Maybe add a comment like "can take hours if latest state size occupies tens of GBs"?

**Review by Longarithm — APPROVED** [review]
LGTM, leaving some readability suggestions:

File: tools/flat-storage/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+near-epoch-manager = { path = "../../chain/epoch-manager" }
+near-primitives = { path = "../../core/primitives" }
+near-store = { path = "../../core/store" }
+nearcore = { path = "../../nearcore" }
+
+[features]
+protocol_feature_flat_state = ["nearcore/protocol_feature_flat_state",
+"near-chain/protocol_feature_flat_state",
+"near-store/protocol_feature_flat_state",]
```
[the remark this answers is not in the document]
Comment (mm-near) [reply tools/flat-storage/Cargo.toml:27]: done.

File: tools/flat-storage/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#[cfg(feature = "protocol_feature_flat_state")]
+pub mod flat;
```
[the remark this answers is not in the document]
Comment (mm-near) [reply tools/flat-storage/src/lib.rs:2]: ok

**Pull request merged**